feature/google/src/main/java/com/revenuecat/purchases/google/findBestOffer.kt

```diff
@@ -0,0 +1,62 @@
+@file:Suppress("MaximumLineLength", "MaxLineLength")
+package com.revenuecat.purchases.google
+
+import com.revenuecat.purchases.models.GoogleSubscriptionOption
+import com.revenuecat.purchases.models.PricingPhase
+
+fun List<GoogleSubscriptionOption>.findBestOffer(): GoogleSubscriptionOption? {
+    val basePlan = this.firstOrNull { it.isBasePlan } ?: return null
+
+    val validOffers = this
+        .filter { !it.isBasePlan }
+        .filter { !it.tags.contains("rc-ignore-best-offer") }
+
+    return findLongestFreeTrial(validOffers) ?: findLowestNoneFreeOffer(validOffers) ?: basePlan
+}
+
+private fun findLongestFreeTrial(offers: List<GoogleSubscriptionOption>): GoogleSubscriptionOption? {
+    return offers.mapNotNull { offer ->
+        offer.freePricingPhase()?.let { pricingPhase ->
+            Pair(offer, parseBillPeriodToDays(pricingPhase.billingPeriod))
+        }
+    }.maxByOrNull { it.second }?.first
+}
+
+private fun findLowestNoneFreeOffer(offers: List<GoogleSubscriptionOption>): GoogleSubscriptionOption? {
+    return offers.mapNotNull { offer ->
+        offer.nonFreePricingPhase()?.let { pricingPhase ->
+            Pair(offer, pricingPhase.priceAmountMicros)
+        }
+    }.minByOrNull { it.second }?.first
+}
+
+private fun GoogleSubscriptionOption.freePricingPhase(): PricingPhase? {
+    return pricingPhases.firstOrNull()?.takeIf { it.priceAmountMicros == 0L }
+}
+
+private fun GoogleSubscriptionOption.nonFreePricingPhase(): PricingPhase? {
+    return pricingPhases.firstOrNull()?.takeIf { it.priceAmountMicros > 0L }
+}
+
+private const val DAYS_IN_YEAR = 365
+private const val DAYS_IN_MONTH = 30
+private const val DAYS_IN_WEEK = 7
+
+// Would use Duration.parse but only available API 26 and up
+internal fun parseBillPeriodToDays(period: String): Int {
+    // Takes from https://stackoverflow.com/a/32045167
+    val regex = "^P(?!\$)(\\d+(?:\\.\\d+)?Y)?(\\d+(?:\\.\\d+)?M)?(\\d+(?:\\.\\d+)?W)?(\\d+(?:\\.\\d+)?D)?(T(?=\\d)(\\d+(?:\\.\\d+)?H)?(\\d+(?:\\.\\d+)?M)?(\\d+(?:\\.\\d+)?S)?)?\$"
```

Comment:
Minor: Can we remove the whole Timing part of the regex? I don't think we can ever have it, and actually ignore it...
So everything after `T`
```suggestion
    val regex = "^P(?!\$)(\\d+(?:\\.\\d+)?Y)?(\\d+(?:\\.\\d+)?M)?(\\d+(?:\\.\\d+)?W)?(\\d+(?:\\.\\d+)?D)?\$"
```